src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -188,6 +188,31 @@ object SnowflakeTelemetry {
     )
   }
 
+  private[snowflake] def sendQueryResult(conn: Connection,
+                                         operation: String,
+                                         query: String,
+                                         queryId: String,
+                                         queryResult: String,
+                                         elapse: Long,
+                                         message: String,
+                                         details: String
+                                         ): Unit = {
+    val metric: ObjectNode = mapper.createObjectNode()
+    metric.put(TelemetryQueryResultFields.SPARK_CONNECTOR_VERSION, Utils.VERSION)
+    metric.put(TelemetryQueryResultFields.OPERATION, operation)
+    metric.put(TelemetryQueryResultFields.QUERY, query)
+    metric.put(TelemetryQueryResultFields.QUERY_ID, queryId)
+    metric.put(TelemetryQueryResultFields.QUERY_RESULT, queryResult)
```

Comment:
queryResult and message are independent fields, but it seems that message is only used for exception conditions. Do you ever expect 'success', but also with a populated message field?

Reply:
Yes. I intend to make it to be easy to support success query.  In the beginning, the message is named as "spark_query_fail". I change it to "spark_query_result" with the QUERY_RESULT field.
The high level idea is for SC dashboard. If we send success message for read and write. we can monitor each customer to see how many spark job they run, how many fail/succeed, and look into details for the fail reason.
We can discuss in the later meeting.